pkg/trace/pb/trace.go

```diff
@@ -33,7 +33,18 @@ func (p *TracerPayload) Cut(i int) *TracerPayload {
 	if i > len(p.Chunks) {
 		i = len(p.Chunks)
 	}
-	new := *p
+	new := TracerPayload{
```

Comment:
This here (introduced by me) looks like another potential source for a performance regression (multiple reads on `p` I'm not sure if the compiler would optimize the copy with the one-line assignment)..... Unfortunately the mutex in the TracerPayload struct doesn't really allow us to do the:
```
new := *p
```
That's flagged by the linters, and makes total sense, copying a mutex makes no sense. Let's look into this if we still have bad performance degradation.

Reply:
Maybe we should use [`Clone`](https://pkg.go.dev/google.golang.org/protobuf/proto#Clone) for this.